### PR TITLE
chore: move `has_methods_*` fields to `/smart-contracts` endpoint response

### LINKED ISSUE #9578: API v2: Read/write for twin contracts
- move `has_methods_(read/read_proxy/write/write_proxy)` from `/addresses` to `/smart-contracts`
- make sure that it returns contract methods for twins

### PR DESCRIPTION
**Pull request by fedor-ivn**
Closes #9578.

### Enhancements

Move `has_methods_*` and `has_custom_methods_*` fields from `/api/v2/addresses/{address_hash}` to `/api/v2/smart-contracts/{address_hash}` endpoint response.

### Bug Fixes

Fix tests to expect calls to rpc node.

## Checklist for your Pull Request (PR)

  - [ ] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/block_scout_web/test/block_scout_web/controllers/api/v2/smart_contract_controller_test.exs [hunk 1]
```diff
@@ -116,6 +122,8 @@ defmodule BlockScoutWeb.API.V2.SmartContractControllerTest do
         "license_type" => "none"
       }
 
+      get_eip1967_implementation_non_zero_address()
```
[the remark this answers is not in the document]
Comment (nikitosing) [reply apps/block_scout_web/test/block_scout_web/controllers/api/v2/smart_contract_controller_test.exs:125]: It is ok

File: apps/block_scout_web/lib/block_scout_web/views/api/v2/smart_contract_view.ex [hunk 2]
```diff
@@ -200,7 +212,7 @@ defmodule BlockScoutWeb.API.V2.SmartContractView do
     |> Map.merge(bytecode_info(address))
   end
 
-  def prepare_smart_contract(address) do
+  def prepare_smart_contract(address, _) do
     bytecode_info(address)
   end
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/views/api/v2/smart_contract_view.ex:217]: If smart contract is unverified, we still need to append 
`has_custom_methods_read`
`has_custom_methods_write`

**Review by vbaranov — COMMENTED** [review]
The PR named as "move has_methods_* fields to /smart-contracts endpoint response". I expected from this naming, that `has_methods_` will be removed from address view and moved to smart contract view.
Do we still need those `has_methods_` in address view?

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/smart_contract_controller.ex [hunk 3]
```diff
@@ -39,7 +39,7 @@ defmodule BlockScoutWeb.API.V2.SmartContractController do
            {:not_found, Chain.find_contract_address(address_hash, @smart_contract_address_options, false)} do
       conn
       |> put_status(200)
-      |> render(:smart_contract, %{address: address})
+      |> render(:smart_contract, %{address: address, conn: conn})
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/smart_contract_controller.ex:42]: It's not necessary to put connection. It is done automatically
```suggestion
      |> render(:smart_contract, %{address: address})
```

Comment (fedor-ivn) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/smart_contract_controller.ex]: Fixed in https://github.com/blockscout/blockscout/pull/9599/commits/3bdd76cbd93910f1c1f3378c02238a8f6396ced7

**Pull request merged**